sermant-plugins/sermant-flowcontrol/flowcontrol-service/src/main/java/io/sermant/flowcontrol/res4j/chain/handler/XdsRateLimitRequestHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.flowcontrol.res4j.chain.handler;
+
+import io.sermant.core.service.xds.entity.FractionalPercent;
+import io.sermant.core.service.xds.entity.XdsRateLimit;
+import io.sermant.core.service.xds.entity.XdsTokenBucket;
+import io.sermant.core.utils.StringUtils;
+import io.sermant.flowcontrol.common.config.CommonConst;
+import io.sermant.flowcontrol.common.entity.BusinessEntity;
+import io.sermant.flowcontrol.common.xds.handler.XdsFlowControlHandler;
+import io.sermant.flowcontrol.common.xds.ratelimit.RateLimitManager;
+import io.sermant.flowcontrol.res4j.chain.AbstractChainHandler;
+import io.sermant.flowcontrol.res4j.chain.HandlerConstants;
+import io.sermant.flowcontrol.res4j.chain.context.RequestContext;
+import io.sermant.flowcontrol.res4j.exceptions.RateLimitException;
+import io.sermant.flowcontrol.res4j.util.RandomUtil;
+
+import java.util.Optional;
+
+/**
+ * Current limiting request handler
+ *
+ * @author zhp
+ * @since 2024-12-05
+ */
+public class XdsRateLimitRequestHandler extends AbstractChainHandler {
+    @Override
+    public void onBefore(RequestContext context, BusinessEntity businessEntity) {
+        handler(businessEntity);
+        super.onBefore(context, businessEntity);
+    }
+
+    private void handler(BusinessEntity businessEntity) {
+        Optional<XdsRateLimit> xdsRateLimitOptional = XdsFlowControlHandler.INSTANCE.getRateLimit(
+                businessEntity.getServiceName(), businessEntity.getRouteName(), businessEntity.getClusterName());
+        if (!xdsRateLimitOptional.isPresent()) {
+            return;
+        }
+        XdsRateLimit xdsRateLimit = xdsRateLimitOptional.get();
+        XdsTokenBucket tokenBucket = xdsRateLimit.getTokenBucket();
+        if (tokenBucket == null || xdsRateLimit.getPercent() == null
+                || tokenBucket.getMaxTokens() == 0 || tokenBucket.getFillInterval() == 0) {
```

Comment:
Check whether tokenBucket.getMaxTokens() can be set to 0 through the crd file of the istio.

Reply:
done